packages/table/src/locator.ts

```diff
@@ -101,28 +101,35 @@ export class Locator implements ILocator {
     }
 
     public getWidestVisibleCellInColumn(columnIndex: number): number {
-        const cells = this.cellContainerElement.getElementsByClassName(Classes.columnCellIndexClass(columnIndex));
-        let max = 0;
-        for (let i = 0; i < cells.length; i++) {
-            const contentWidth = Utils.measureElementTextContent(cells.item(i)).width;
+        const columnCellSelector = this.getColumnCellSelector(columnIndex);
+        const columnHeaderAndBodyCells = this.tableElement.querySelectorAll(columnCellSelector);
+
+        let maxWidth = 0;
+        for (let i = 0; i < columnHeaderAndBodyCells.length; i++) {
+            const contentWidth = Utils.measureElementTextContent(columnHeaderAndBodyCells.item(i)).width;
             const cellWidth = Math.ceil(contentWidth) + Locator.CELL_HORIZONTAL_PADDING * 2;
-            if (cellWidth > max) {
-                max = cellWidth;
+            if (cellWidth > maxWidth) {
+                maxWidth = cellWidth;
             }
         }
-        return max;
+        return maxWidth;
     }
 
     public getTallestVisibleCellInColumn(columnIndex: number): number {
-        const cells = this.cellContainerElement.getElementsByClassName(
-            `${Classes.columnCellIndexClass(columnIndex)} ${Classes.TABLE_CELL}`,
-        );
-        let max = 0;
-        for (let i = 0; i < cells.length; i++) {
-            const cellValue = cells.item(i).querySelector(`.${Classes.TABLE_TRUNCATED_VALUE}`);
-            const cellTruncatedFormatText = cells.item(i).querySelector(`.${Classes.TABLE_TRUNCATED_FORMAT_TEXT}`);
-            const cellTruncatedText = cells.item(i).querySelector(`.${Classes.TABLE_TRUNCATED_TEXT}`);
+        // consider only body cells, hence the extra Classes.TABLE_CELL specificity
+        const columnCellSelector = this.getColumnCellSelector(columnIndex);
+        const columnBodyCells = this.tableElement.querySelectorAll(`${columnCellSelector}.${Classes.TABLE_CELL}`);
+
+        let maxHeight = 0;
+        for (let i = 0; i < columnBodyCells.length; i++) {
```

Comment:
I realize we were doing this with a for loop before -- but what exactly is the structure returned from the querySelectorAll ? Can we use some form of iterator or foreach instead?

Reply:
Seems like we don't have typings for any of this stuff: https://developer.mozilla.org/en-US/docs/Web/API/NodeList. So need to leave as is.